omegaconf/_utils.py

```diff
@@ -349,11 +349,8 @@ def _is_missing_value(value: Any) -> bool:
 
 
 def _is_missing_literal(value: Any) -> bool:
-    from omegaconf import MISSING
-
-    ret = isinstance(value, str) and value == MISSING
-    assert isinstance(ret, bool)
-    return ret
+    # Uses literal '???' instead of the MISSING const for performance reasons.
+    return isinstance(value, str) and value == "???"
```

Comment:
We can make it 25% faster (on top of this function) by throwing away the instance check, but that's conflicting with the dataframe issue (a test will fail).

At this point, we can also consider caching the value type on the node, to avoid recalculating it more than once.